services/twilio_verify/twilio_verify.js

```diff
@@ -0,0 +1,52 @@
+const twilioClient = require('twilio');
+
+const ACCOUNT_SID = process.env.TWILIO_ACCOUNT_SID;
+const AUTH_TOKEN = process.env.TWILIO_AUTH_TOKEN;
+const VERIFY_SERVICE_SID = process.env.TWILIO_VERIFY_SERVICE_SID;
+
+const VERIFY_DELIVERY_CHANNEL = 'sms';
+
+// https://www.twilio.com/docs/api/errors/XXXXX
+const ERROR_RESPONSES_BY_CODE = {
+    20404: { status: 429, text: '2FA request expired, please try again' },
+    20429: { status: 429, text: '2FA rate limit reached, please try again later' },
+    60202: { status: 429, text: '2FA verification check limit reached, please try again later' },
+    60203: { status: 429, text: '2FA delivery limit reached, please try again later' },
+};
+
+const DEFAULT_ERROR_RESPONSE = {
+    status: 500,
+    text: 'Twilio Verify Error',
+};
+
+module.exports = class TwilioVerifyService {
+    constructor({
+        twilio = twilioClient,
+    } = {}) {
+        const client = twilio(ACCOUNT_SID, AUTH_TOKEN);
+
+        this.verifyService = client
+            .verify
+            .services(VERIFY_SERVICE_SID);
+    }
+
+    handleTwilioErrors(fn) {
```

Comment:
I think this works well with `send` which is a one liner, but not so much with `verify`:
```js
const { valid } = await this.verifyService.verificationChecks.create({ to, code })
    .catch((error) => {
        this.handleTwilioErrors(error);
    });
return valid;
```
I was sorta taking inspiration from `bluebird`s `Promise.try()` here, I like the idea of wrapping everything as it gives us the flexibility to compose many twilio methods and `catch` all errors which may be thrown from the individual methods. With this approach we'd need to apply to all methods individually or similarly wrap everything.

That was my reasoning but i'm on the fence, what are your thoughts? :)

Reply:
With regards to stacktrace node still picks up the correct method names:
![image](https://user-images.githubusercontent.com/11974624/160299419-28bf654b-abcc-477e-9e97-676cc2cbb53b.png)

Reply:
For `verify` I think a `try`/`catch` would work pretty well 🙂 

I see your point about `Promise.try`, though it still seems a bit of a heavy abstraction in the current implementation since the two methods using it are only calling a single method capable of throwing a relevant error. Coupled with the fact that it's not doing anything more complex than augmenting the error it struck me as a bit unconventional.

But I think I'm getting more hung up on the syntax and low sample size, when I think of the behavior as more of a decorator (e.g. `@handleTwilioErrors()`) it seems like a reasonable way to encapsulate the common error logic in a composable way.